src/borg/archive.py

```diff
@@ -468,6 +470,8 @@ def save(self, name=None, comment=None, timestamp=None, additional_metadata=None
         data = self.key.pack_and_authenticate_metadata(metadata.as_dict(), context=b'archive')
         self.id = self.key.id_hash(data)
         self.cache.add_chunk(self.id, Chunk(data), self.stats)
+        self.repository.async_wait()
```

Comment:
Yes, a window on responses would do the job. Most code pathes could work with a window of 1 (just enough to detect errors). Maybe make window a non defaulted parameter so this shows up in reviews more easily?
My other idea which was less orthogonal was to instead have a async_wait_and_get_response method that would just be window=1 and combines async_response into it. Less pure, but maybe easier to use.